addons/binding/org.openhab.binding.oceanic/src/main/java/org/openhab/binding/oceanic/handler/OceanicThingHandler.java

```diff
@@ -0,0 +1,216 @@
+/**
+ * Copyright (c) 2014 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.oceanic.handler;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.Type;
+import org.eclipse.smarthome.core.types.TypeParser;
+import org.openhab.binding.oceanic.OceanicBindingConstants.OceanicChannelSelector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OceanicHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Karel Goderis - Initial contribution
+ */
+public class OceanicThingHandler extends SerialThingHandler {
+
+    // List of Configuration constants
+    public static final String INTERVAL = "interval";
+
+    private Logger logger = LoggerFactory.getLogger(OceanicThingHandler.class);
+
+    private ScheduledFuture<?> pollingJob;
+    private String lastLineReceived = "";
+    private long GRACE_PERIOD = 1000;
+    private long REQUEST_TIMEOUT = 15000;
+
+    public OceanicThingHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing Oceanic handler.");
+        if (pollingJob != null && !pollingJob.isCancelled()) {
+            pollingJob.cancel(true);
+            pollingJob = null;
+        }
+        super.dispose();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Oceanic handler.");
+
+        if (getConfig().get(BAUD_RATE) == null) {
```

Comment:
This config param can not be user set but is set during initialize(). The reason for this is that the handler extends the SerialThingHandler class (which, because of lack of a serial binding, is included in the oceanic bundle), for which BAUD_RATE and BUFFER_SIZE are relevant.